controllers/release/adapter.go

```diff
@@ -681,8 +933,8 @@ func (a *adapter) validatePipelineRef() *controller.ValidationResult {
 	return &controller.ValidationResult{Valid: true}
 }
 
-// validateSinglePipeline checks that the Pipeline is defined exclusively in the ReleasePlan or in the ReleasePlanAdmission.
-func (a *adapter) validateSinglePipeline() *controller.ValidationResult {
+// validatePipelineDefined checks that a Pipeline is defined in either the ReleasePlan or in the ReleasePlanAdmission.
+func (a *adapter) validatePipelineDefined() *controller.ValidationResult {
```

Comment:
I don't think this function should consider the final pipeline. Final pipeline is called final because it is like a finally block in code. You wouldn't have it be the only pipeline defined